### PR TITLE
[jsk_fetch_startup] [jsk_robot_startup] Use demo-based mongodb logging in kitchen-demo

### PR DESCRIPTION
**Pull request by tkmtnt7000**
With this PR, we can store `images`, `speech_to_text`, `sound_play/goal` in kitchen-demo.
We can get these data from `jsk_robot_lifelog/go_to_kitchen` data collection

### CODE REVIEW

File: jsk_robot_common/jsk_robot_startup/lifelog/mongodb.launch [hunk 1]
```diff
@@ -18,6 +19,9 @@
   <!-- Replication -->
   <group if="$(arg replicate)">
     <rosparam command="load" file="$(arg replicator_param_path)" />
+    <rosparam subst_value="true" if="$(eval arg('extra_collections') != '')">
+      extra_collections: $(arg extra_collections)
+    </rosparam>
     <node name="replication_client"
           pkg="jsk_robot_startup" type="periodic_replicator_client.py"
           output="screen" machine="$(arg machine)"/>
```
Comment (knorth55) [inline jsk_robot_common/jsk_robot_startup/lifelog/mongodb.launch:27]: `rosparam` should be set in `node` section, isn't it?

```suggestion

    <node name="replication_client"
          pkg="jsk_robot_startup" type="periodic_replicator_client.py"
          output="screen" machine="$(arg machine)">
      <rosparam subst_value="true" if="$(eval arg('extra_collections') != '')">
        extra_collections: $(arg extra_collections)
      </rosparam>
    </node>
```

**Review by knorth55 — CHANGES_REQUESTED** [review]

File: jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch [hunk 2]
```diff
@@ -277,6 +280,32 @@
...
+          name="dialogflow_logger"
+          pkg="jsk_robot_startup" type="mongo_record.py"
+          machine="$(arg machine)"
+          respawn="$(arg respawn)">
+      <rosparam subst_value="true">
+        topics:
+        - /dialogflow_client/text_action/goal
+        - /dialogflow_client/text_action/result
+      </rosparam>
+    </node>
```
Comment (knorth55) [inline jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch:307]: add `collection` arg in these nodes, too.

**Review by knorth55 — CHANGES_REQUESTED** [review]

File: jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch [hunk 3]
```diff
@@ -310,6 +339,22 @@
...
+          respawn="$(arg respawn)">
+      <rosparam subst_value="true">
+        database: $(arg database)
+        topics:
+        - /edgetpu_human_pose_estimator/output/poses
+      </rosparam>
+      <rosparam subst_value="true" if="$(eval arg('collection') != '')">
+        collection: $(arg collection)
+      </rosparam>
+    </node>
```
Comment (knorth55) [inline jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch:356]: can you also add logger fot `edgetpu_object_detector/output` too?

File: jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch [hunk 4]
```diff
@@ -310,6 +339,22 @@
...
+    <!-- human pose logger -->
+    <node if="$(arg save_human_pose)"
+          name="human_pose_logger"
+          pkg="jsk_robot_startup" type="mongo_record.py"
+          machine="$(arg machine)"
+          respawn="$(arg respawn)">
+      <rosparam subst_value="true">
+        database: $(arg database)
+        topics:
+        - /edgetpu_human_pose_estimator/output/poses
```
Comment (knorth55) [inline jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch:351]: you can log all the messages like `~output/class`.
logging `~output/poses` only may not be enough.

https://github.com/jsk-ros-pkg/coral_usb_ros/blob/26cba3835b8b13d92738cb7ca3c76e7164f323a0/python/coral_usb/human_pose_estimator.py#L53-L59

**Review by knorth55 — CHANGES_REQUESTED** [review]

**Comment by knorth55** [comment]
I checked `mongodb` database, but the `images` are not found in 2022/11/11.
can you correctly store `images`?

also, can you store `compressedDepth` too, for point cloud?

**Comment by knorth55** [comment]
please rebase again, sorry.

File: jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch [hunk 5]
```diff
@@ -295,15 +333,48 @@
...
+    <group if="$(arg save_object_detection)">
+      <node if="$(arg save_edgetpu_object_detector)"
+            name="object_detection_logger"
+            pkg="jsk_robot_startup" type="mongo_record.py"
+            machine="$(arg machine)"
+            respawn="$(arg respawn)">
+        <rosparam subst_value="true">
+          database: $(arg database)
+          topics:
+          - /edgetpu_human_pose_estimator/output/class
```
Comment (knorth55) [inline jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch:345]: https://github.com/jsk-ros-pkg/coral_usb_ros#object-detector-edgetpu_object_detectorpy-1

please add `output/rects`

Comment (knorth55) [reply jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch:345]: typo: human -> object detector?

**Review by knorth55 — CHANGES_REQUESTED** [review]

File: jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch (hunk 2 above)
Comment (tkmtnt7000) [reply jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch]: Thank you for reviewing. I added in [dd0061f](https://github.com/jsk-ros-pkg/jsk_robot/pull/1728/commits/dd0061f19c70db31a2d5eca6d55254a58ca8ef43).

File: jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch (hunk 5 above)
Comment (tkmtnt7000) [reply jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch]: Thank you very much for reviewing. 
I fixed it in [d19109a](https://github.com/jsk-ros-pkg/jsk_robot/pull/1728/commits/d19109a730519f57e463801d6d86f08e3f30d8a7) and [d034ac9](https://github.com/jsk-ros-pkg/jsk_robot/pull/1728/commits/d034ac969beff94b096e5af46fe8e0c3827e7fac).

File: jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch (hunk 3 above)
Comment (tkmtnt7000) [reply jsk_robot_common/jsk_robot_startup/lifelog/common_logger.launch]: Thank you. I added.

**Review by knorth55 — APPROVED** [review]

**Comment by knorth55** [comment]
It seems it works correctly.
With this launch, we can collect data to `jsk_robot_lifelog.go_to_kitchen` when runnning the kitchen demo.

Screenshot from 2022-12-01 21-58-55

**Pull request merged**